packages/@react-aria/utils/src/runAfterTransition.ts

```diff
@@ -26,7 +26,14 @@ function setupGlobalEvents() {
     return;
   }
 
-  let onTransitionStart = (e: TransitionEvent) => {
+  function isTransitionEvent(event: Event): event is TransitionEvent {
+    return 'propertyName' in event;
+  }
```

Comment:
Added `isTransitionEvent` helper because when applying `transitioncancel` for `event.target` the actual type of the event is `Event` which is the base-class of `TransitionEvent`.  This helps preparing the module for strict-mode in the future.

Reply:
I was also thinking about using `instanceof` here but I wasn't quite sure if it was supported for events

Reply:
Probably not needed, we don't make use of anything that isn't available on Event, so could just make the handlers accept Event until we have a need. They are implementation details anyways. Not sure how unique that property is, but this won't hurt right now.